assembly/src/assembler/instruction/u32_ops.rs

```diff
@@ -393,61 +273,48 @@ pub fn u32rotr(
     op_mode: U32OpMode,
     imm: Option<u8>,
 ) -> Result<Option<CodeBlock>, AssemblyError> {
-    match imm {
-        None => match op_mode {
-            U32OpMode::Checked => {
+    match (imm, op_mode) {
+        (Some(imm), U32OpMode::Checked) => {
+            // Assert the top of the stack is a u32 value.
+            // NOTE: We cannot use U32Assert2 since we are potentially pushing a number larger
+            // than u32 for b.
+            // TODO: double check this logic: we shouldn't use U32MUL if both inputs are not
+            // guaranteed to be u32 values
+            validate_param(imm, 0, 31)?;
+            span.push_ops([Pad, U32assert2, Drop, Push(Felt::new(1 << (32 - imm)))]);
+        }
+        (Some(imm), U32OpMode::Unchecked) => {
+            validate_param(imm, 0, 31)?;
```

Comment:
nit: similar to the other comment - let's replace `31` with a constant from `lib.rs`.

Reply:
Fixed